### DISCUSSION TITLE
Strapi sendGrid on Heroku - Bad Request - Auth.form.error.email.format

### DISCUSSION THREAD

**Discussion by elirodrigues** [strapi/strapi#6034] [Legacy Topics]
Hey guys,
I'm looking for help to complete the configuration of SendGrid plugin. Currently, my SendGrid is provided by Heroku. But, it keeps insistently sending me the message errror: "Auth.form.error.email.format".

Bellow the steps I've followed:

1. (locally) Install Strapi and configure contents, users, etc 
2. (locally) npm install strapi-provider-email-sendgrid@alpha --save
3. (locally) npm run develop --watch-admin
4. (Heroku) go to Heroku and provide SendGrid
5. (through Heroku) On Sendgrid: generate SendGrid API key
6. (locally) Go to Strapi Admin / Settings / E-mail 
7. (locally) Select Sendgrid as a provider and add Sendgrid API Key
8. (locally) Go to Groups and Permissions and allow everything to Email plugin
9. (locally) Execute a POST on Postman

**Screenshots**

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/25332847/80894005-68944600-8cad-11ea-867b-4570042a9c36.png">

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/25332847/80894021-85c91480-8cad-11ea-9fae-257a17a02aeb.png">

![image](https://user-images.githubusercontent.com/25332847/80894112-7ac2b400-8cae-11ea-97b9-1f7498129a13.png)

**Code snippets**
I've also tested the connection using Python, just for the check and it works fine:

`# using SendGrid's Python Library
import os
from sendgrid import SendGridAPIClient
from sendgrid.helpers.mail import Mail

message = Mail(
    from_email='MYEMAIL@gmail.com',
    to_emails='MYEMAIL@gmail.com',
    subject='Consegui - Sending with Twilio SendGrid is Fun',
    html_content='<strong>and easy to do anywhere, even with Python</strong>')
try:
    sg = SendGridAPIClient(os.environ.get('MY_SENDGRID_KEY'))
    response = sg.send(message)
    print(response.status_code)
    print(response.body)
    print(response.headers)
except Exception as e:
    print(e)`

**System**
- Strapi version: 
- Node.js version:node v12.16.2
- NPM version:6.14.4
- Strapi version: 3.0.0-beta.20
- Database: MongoDb in Mongo Atlas, via mongoose
- Operating system: Mac OS Catalina 10.15.4

**Additional context**
With all that being said. Is there any advice from the community?

Tks, Eli.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by elirodrigues** [comment]
Ok, thank you!

Em dom., 3 de mai. de 2020 às 04:13, DMehaffy <notifications@github.com>
escreveu:

**Reply by elirodrigues** [reply]
I've found that it was necessary to Allow the "forgot-password" on the Admin Panel:

![image](https://user-images.githubusercontent.com/25332847/80929648-1d993200-8d84-11ea-8ec6-8a3edfa5f106.png)

After doing this, I used Postman to test both with a valid and invalid e-mail.

The answer for an invalid email is:
![image](https://user-images.githubusercontent.com/25332847/80929670-50dbc100-8d84-11ea-9e5a-dfad2bb9bcdb.png)

The answer for a valid one:
![image](https://user-images.githubusercontent.com/25332847/80929685-6bae3580-8d84-11ea-9b2e-d2a907164219.png)

My last possible doubt would be related to this dialog:
![image](https://user-images.githubusercontent.com/25332847/80929715-99937a00-8d84-11ea-9081-f86b9bba7dd5.png)

What am I supposed to mark as an option? It doesn't work through Postman, but works directly in the admin panel

![image](https://user-images.githubusercontent.com/25332847/80929822-60a7d500-8d85-11ea-91e4-f0233b0525e7.png)

![image](https://user-images.githubusercontent.com/25332847/80929829-6f8e8780-8d85-11ea-916d-db904e94edc8.png)

As you can check in the last image, it has used SendGrid properly.

Please, advice: how to make it work for my front-end application? any other item to be allowed?

**Reply by elirodrigues** [reply]
Finally, I've marked every checkbox existent in the admin panel, whitelisted all IPs in Sendgrid and it has worked.
